numpy/testing/utils.py

```diff
@@ -24,5 +25,5 @@
         'assert_allclose', 'IgnoreException', 'clear_and_catch_warnings',
         'SkipTest', 'KnownFailureException', 'temppath', 'tempdir', 'IS_PYPY',
         'HAS_REFCOUNT', 'suppress_warnings', 'assert_array_compare',
-        '_assert_valid_refcount', '_gen_alignment_data', 'assert_no_gc_cycles'
```

Comment:
There seem to be a number of objects within `__all__` with a rather questionable public status, these two being the most obvious ones. Deprecation might be overkill, but doing a review and removing some of the offending objects from `__all__` would be a good start.

Reply:
Note that we will remove `numpy/testing/utils.y` at some point. It there for backwards compatibility and to warn downstream to change the import.